starfish/test/test_utils.py

```diff
@@ -70,3 +72,171 @@ def codebook_array_factory() -> Codebook:
         },
     ]
     return Codebook.from_code_array(data)
+
+
+def _create_dataset(
+    pixel_dimensions: Tuple[int, int, int],
+    spot_coordinates: Sequence[Tuple[int, int, int]],
+    codebook: Codebook
+) -> ImageStack:
+    """
+    creates a numpy array containing one spot per codebook entry at spot_coordinates. length of
+    spot_coordinates must therefore match the number of codes in Codebook.
+    """
+    assert len(spot_coordinates) == codebook.sizes[Features.TARGET]
+
+    data_shape = (
+        codebook.sizes[Axes.ROUND.value],
+        codebook.sizes[Axes.CH.value],
+        *pixel_dimensions
+    )
+    imagestack_data = np.zeros((data_shape), dtype=np.float32)
+
+    for ((z, y, x), f) in zip(spot_coordinates, range(codebook.sizes[Features.TARGET])):
+        imagestack_data[:, :, z, y, x] = codebook[f].transpose(Axes.ROUND.value, Axes.CH.value)
+
+    # blur with a small non-isotropic kernel TODO make kernel smaller.
+    imagestack_data = gaussian_filter(imagestack_data, sigma=(0, 0, 0.7, 1.5, 1.5))
+    return ImageStack.from_numpy_array(imagestack_data)
+
+
+def two_spot_one_hot_coded_data_factory() -> Tuple[Codebook, ImageStack, float]:
+    """
+    Produce a 2-channel 2-round Codebook with two codes and an ImageStack containing one spot from
```

Comment:
what does the word hot refer to here?

Reply:
Sorry for missing this comment. One-hot is a type of codebook encoding. 

https://en.wikipedia.org/wiki/One-hot